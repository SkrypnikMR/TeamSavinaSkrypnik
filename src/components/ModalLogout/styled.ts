import styled from 'styled-components';

export const StModalLogout = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
`;
