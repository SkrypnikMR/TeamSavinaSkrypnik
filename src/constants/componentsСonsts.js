export const regInputs = [{
    id: 'email',
    label: 'email_label',
    placeholder: 'email_placeholder',
},
{
    id: 'password',
    label: 'password_label',
    placeholder: 'password_placeholder',
},
{
    id: 'confirm',
    label: 'confirm_label',
    placeholder: 'confirm_placeholder',
},
{
    id: 'firstName',
    label: 'first_name_label',
    placeholder: 'first_name_placeholder',
},
{
    id: 'lastName',
    label: 'last_name_label',
    placeholder: 'last_name_placeholder',
}];
export const logInputs = [{
    id: 'email',
    label: 'email_label',
    placeholder: 'email_placeholder',
},
{
    id: 'password',
    label: 'password_label',
    placeholder: 'password_placeholder',
}];
export const HEADER_CONTROL_BTNS = [
    {
        id: 'notif_btn',
        value: '1',
        content: 'üîï',
    },
    {
        id: 'notif_btn',
        value: '',
        content: 'üîî',
    },
    {
        id: 'theme_btn',
        value: 'light',
        content: 'üåö',
    },
    {
        id: 'theme_btn',
        value: 'dark',
        content: 'üåû',
    },
    {
        id: 'en',
        content: 'english_translate_btn',
        value: 'en',
    },
    {
        id: 'ru',
        content: 'russian_translate_btn',
        value: 'ru',
    },
    {
        id: 'ar',
        content: 'arabian_translate_btn',
        value: 'ar',
    },
    {
        id: 'account',
        content: 'üë®‚Äçüíª',
        rout: '/account',
        value: 'account',
    },
    {
        id: 'logOut',
        content: 'üö™',
        rout: '/',
        value: 'logOut',
    }];
export const INPUT__MESSAGE = [{
    id: 'message',
    placeholder: 'placeholder_control_input',
}];
export const INPUT_PLACEHOLDER_SEARCH = [
    {
        id: 'search',
        placeholder: 'placeholder_shearch_input',
    },
];
export const MY_ACCOUNT_EDIT = [{
    id: 'firstName',
    label: 'first_name_label',
    name: 'firstName',
    value: 'first_name_update',
},
{
    id: 'lastName',
    label: 'last_name_label',
    name: 'lastName',
    value: 'last_name_update',
},
{
    id: 'age',
    label: 'age',
    name: 'age',
    value: 'age',
},
{
    id: 'city',
    label: 'city',
    name: 'city',
    value: 'city',
},
{
    id: 'company',
    label: 'company',
    name: 'company',
    value: 'company',
},
{
    id: 'hobby',
    label: 'hobby',
    name: 'hobby',
    value: 'hobby',
},
];
export const MODAL_USERS_BUTTON = [{
    id: 'closeModal',
    content: 'modal_users_cancel',
    value: '',
},
{
    id: 'sendMessage',
    content: 'modal_users_confirm',
    value: '',
}];
