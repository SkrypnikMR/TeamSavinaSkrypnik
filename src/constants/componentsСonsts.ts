export const regInputs = [{
    id: 'login',
    label: 'login_label',
    placeholder: 'login_placeholder',
},
{
    id: 'password',
    label: 'password_label',
    placeholder: 'password_placeholder',
},
{
    id: 'confirm',
    label: 'confirm_label',
    placeholder: 'confirm_placeholder',
}];
export const logInputs = [{
    id: 'login',
    label: 'login_label',
    placeholder: 'login_placeholder',
},
{
    id: 'password',
    label: 'password_label',
    placeholder: 'password_placeholder',
}];
export const HEADER_CONTROL_BTNS = [
    {
        id: 'theme_btn',
        value: 'dark',
        content: 'ðŸ’¡',
    },
    {
        id: 'en',
        content: 'english_translate_btn',
        value: 'en',
    },
    {
        id: 'ru',
        content: 'russian_translate_btn',
        value: 'ru',
    }];
export const INPUT__MESSAGE = [{
    id: 'message',
    placeholder: 'placeholder_control_input',
}];
export const INPUT_PLACEHOLDER_SEARCH = [
    {
        id: 'search',
        placeholder: 'placeholder_shearch_input',
    },
];
export const MODAL_USERS_BUTTON = [{
    id: 'closeModal',
    content: 'modal_users_cancel',
    value: '',
},
{
    id: 'sendMessage',
    content: 'modal_users_confirm',
    value: '',
}];
export const GAME_TYPE = {
    tic_tac_toe: 'Tic-tac-toe',
    checkers: 'Checkers',
};

export const INITIAL_STATE_FOR_CHECKER = Array.from({ length: 64 }, (_, i) => {
    let color = null;
    let isChecked = null;
    if (i >= 41 && i <= 64) {
        color = "black";
        if (parseInt((i / 8) + i) % 2 === 0) {
            isChecked = true;
        }
    } else if (i >= 0 && i <= 23) {
        color = "white";
        if (parseInt((i / 8) + i) % 2 === 0) {
            isChecked = true;
        }
    }
    return { position: 63 - i, isChecked, color };
});
