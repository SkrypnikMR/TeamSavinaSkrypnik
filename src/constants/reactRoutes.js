export const APP_ROUTES = {
    registration: '/registration',
    login: '/',
    chat: '/chat',
    account: '/account',
};
