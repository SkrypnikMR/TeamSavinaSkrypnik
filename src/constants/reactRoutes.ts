interface IRoutes {
    registration: string;
    login: string;
}

export const APP_ROUTES: IRoutes = {
    registration: '/registration',
    login: '/',
};
