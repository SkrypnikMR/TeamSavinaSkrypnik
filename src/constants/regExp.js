export const regExp = {
    emailRegExp: /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/,
    passwordRegExp: /^[a-z0-9A-Z]{6,25}/i,
    firstAndLastNameRegExp: /^[a-zA-Za-яА-Я]{2,30}/,
};
