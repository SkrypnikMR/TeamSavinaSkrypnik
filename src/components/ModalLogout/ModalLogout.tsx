import React from 'react';
import { useTranslation } from 'react-i18next';
import Button from '../UI/Button';
import { APP_ROUTES } from '../../constants/reactRoutes';
import { StModalButtonBox } from '../ModalCreateRoom/styled';
import { StModalLogout } from './styled';

const ModalLogout = ({ handlecloseModal, history, exitGame }) => {
    const handleOkClick = () => {
        exitGame();
        history.push(APP_ROUTES.login);
        localStorage.clear();
        handlecloseModal();
    };
    const { t } = useTranslation();
    return (
        <StModalLogout>
            <p>{t('if_logout')}</p>
            <StModalButtonBox>
                <Button 
                    content="OK"
                    focusColor="black"
                    onClick={handleOkClick}
                    id="okButton"
                    name="okButton"
                    value="OK"
                    type="button"
                    title="OK"
                />
                <Button
                    content={t('cancel')}
                    focusColor="black"
                    onClick={handlecloseModal}
                    id="CancelButton"
                    name="CancelButton"
                    value="Cancel"
                    type="button"
                    title="Cancel"
                />
            </StModalButtonBox>
        </StModalLogout>
    );
};

export default ModalLogout;
