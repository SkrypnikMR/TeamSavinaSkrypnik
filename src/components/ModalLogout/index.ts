import { Dispatch } from 'redux';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import ModalLogout from './ModalLogout';
import { exitGame } from '../../store/game/actions';

const mapDispatchToProps = (dispatch: Dispatch) => ({ exitGame: () => dispatch(exitGame()) });

export default connect(null, mapDispatchToProps)(withRouter(ModalLogout));
