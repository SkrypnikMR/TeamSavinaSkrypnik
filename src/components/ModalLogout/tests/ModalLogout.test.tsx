import React from 'react';
import { shallow, mount } from 'enzyme';
import ModalLogout from '../ModalLogout';
import { APP_ROUTES } from '../../../constants/reactRoutes';

describe('ModalCreateRoom', () => {
    let props;
    let modal;
    beforeEach(() => {
        modal = document.createElement('div');
        modal.setAttribute('id', 'modal');
        document.body.append(modal);
    props = {
        handlecloseModal: jest.fn(),
        exitGame: jest.fn(),
        history: { push: jest.fn() },
    };
    });
    it('should snapshot', () => {
        const component = shallow(<ModalLogout {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StModalButtonBox', () => {
        const component = mount(<ModalLogout {...props}/>);
        expect(component.find('styled__StModalButtonBox')).toHaveLength(1);
    });
    it('should click on first button, then call exitGame and handlecloseModal', () => {
        const component = mount(<ModalLogout {...props}/>);
        component.find('Button').at(0).props().onClick();
        expect(props.exitGame).toHaveBeenCalled();
        expect(props.handlecloseModal).toHaveBeenCalled();
        expect(props.history.push).toHaveBeenCalledWith(APP_ROUTES.login);
    });
});
