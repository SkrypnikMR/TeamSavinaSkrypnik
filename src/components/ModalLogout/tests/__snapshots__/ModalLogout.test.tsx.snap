// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCreateRoom should snapshot 1`] = `"<div class=\\"styled__StModalLogout-sc-y5169l-0 dcuLTI\\"><p>if_logout</p><div class=\\"styled__StModalButtonBox-sc-1lbp0ln-0 kgpYEL\\"><button id=\\"okButton\\" name=\\"okButton\\" type=\\"button\\" value=\\"OK\\" title=\\"OK\\" class=\\"styled__StButton-sc-1y35zax-0 fFCnPo\\">OK</button><button id=\\"CancelButton\\" name=\\"CancelButton\\" type=\\"button\\" value=\\"Cancel\\" title=\\"Cancel\\" class=\\"styled__StButton-sc-1y35zax-0 efYwVL\\">cancel</button></div></div>"`;
